_posts/2020-06-04-App Service Continuous Deployment for Windows Containers with GitHub Actions.md

```diff
@@ -0,0 +1,403 @@
+
+### .NET Framework
+
+.NET Framework applications will work best with a multi-stage build.  This example copies over the necessary project files and packages before it creates the publish files for deployment to Azure.
+
+```dockerfile
+# Set the base image
+FROM mcr.microsoft.com/dotnet/framework/sdk:4.8 as build
+WORKDIR "/src"
+
+# Copy packages to your image and restore them
+COPY taskapp/taskapp.sln .
+COPY taskapp/taskapp/taskapp.csproj taskapp/taskapp/taskapp.csproj
+COPY taskapp/taskapp/packages.config taskapp/taskapp/packages.config
+RUN nuget restore taskapp/taskapp/packages.config -PackagesDirectory taskapp/packages
+
+# Add files from source to the current directory and publish the deployment files to the folder profile
+COPY . .
+WORKDIR /src/taskapp/taskapp
+RUN msbuild taskapp.csproj /p:Configuration=Release /m /p:DeployOnBuild=true /p:PublishProfile=FolderProfile
+
+# Layer the production runtime image
+FROM mcr.microsoft.com/dotnet/framework/aspnet:4.8-windowsservercore-ltsc2019 as deploy
+
+# Add the publish files into the right directory
+WORKDIR /inetpub/wwwroot
+COPY --from=build /src/taskapp/taskapp/bin/Release/Publish .
+```
+
+### .NET Core
+
+For .NET Core, the nano server base image must be "1809" to be compatible with what Azure currently supports.  Keep in mind this may change in the future.
+
+```dockerfile
+# Set the base image
+FROM mcr.microsoft.com/dotnet/core/aspnet:3.1-nanoserver-1809 AS base
+WORKDIR /app
+EXPOSE 80
+
+# Add the sdk so you can run the dotnet restore and build commands
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1-nanoserver-1809 AS build
+WORKDIR /src
+COPY *.csproj ./
+RUN dotnet restore "taskapp.csproj"
+COPY . .
+WORKDIR "/src"
+RUN dotnet build "taskapp.csproj" -c Release -o /app/build
+
+# Create the publish files
+FROM build AS publish
+RUN dotnet publish "taskapp.csproj" -c Release -o /app/publish
+
+# Copy the publish files into the container
+FROM base AS final
+WORKDIR /app
+COPY --from=publish /app/publish .
+ENTRYPOINT ["dotnet", "taskapp.dll"]
+
+```
+
+## Create the GitHub Workflow
+
+Now that we have our resources created, secrets secured, and dockerfile in order we can start building our workflow file for continuous deployment.  The workflow file is your yaml file that sits in your repository that contains a collection of actions that run when triggered.  Add the workflow file by going to the *Actions* tab in your repository and *Set up a workflow yourself*.</BR>
+
+![ghactions]({{ site.baseurl }}/media/2020/06/githubactions_1.jpg)
+
+After choosing this option, you'll have starter code that explains how basic GitHub Actions work.  If you are new to this I recommend you read the comments, but we won't be needing any of the starting code so you can remove it.  
+
```

Comment:
If I fork your linked repos, I don't need to do this step right?

Reply:
Correct, you don't need to do this if they are forked since the workflow file is already in the repo.  If they fork the repo, they just need to add in their secrets and run it after.